cassovary-examples/src/main/java/RandomWalkJava.java

```diff
@@ -91,7 +91,7 @@ public static void main(String[] args) {
   @SuppressWarnings("unchecked")
   private static Map<Integer, Integer> scalaIntsMapToJavaMap(scala.collection.Map<Object, Object> map) {
     Map<Object, Object> javaMap = JavaConverters$.MODULE$.mapAsJavaMapConverter(map).asJava();
-    HashMap<Integer, Integer> javaIntsMap = new HashMap<>();
+    HashMap<Integer, Integer> javaIntsMap = new HashMap<Integer, Integer>();
```

Comment:
Which version of java are you using?  To my knowledge that should compile for java 7 and up.  

http://docs.oracle.com/javase/7/docs/technotes/guides/language/type-inference-generic-instance-creation.html